src/hooks/screen/useChatScreenHook.ts

```diff
@@ -214,7 +214,9 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       if (previousChat) {
         if (
           currentChat?.userId === previousChat?.userId &&
-          (previousChat?.rawJson?.isSystem || previousChat?.rawJson?.type === 'system')
+          (previousChat?.rawJson?.isSystem ||
+            previousChat?.rawJson?.type === 'system' ||
+            previousChat?.rawJson?.message?.type === 'system')
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The condition check for `previousChat?.rawJson?.message?.type === 'system'` has been added. This seems to be an additional check to determine if the previous chat message was a system message. However, it's not clear why this check is necessary when there are already checks for `previousChat?.rawJson?.isSystem` and `previousChat?.rawJson?.type === 'system'`. If all these properties indicate the same thing (i.e., whether the message is a system message), then only one of them should be used for consistency and to avoid confusion. If they mean different things, then it would be helpful to add comments explaining their differences.

If they all mean the same thing, you could simplify the code as follows:

```diff
-  (previousChat?.rawJson?.isSystem ||
-   previousChat?.rawJson?.type === 'system' ||
-   previousChat?.rawJson?.message?.type === 'system')
+  (previousChat?.rawJson?.isSystem)
```

If they mean different things, consider adding comments to clarify their purposes.


<!-- This is an auto-generated comment by OSS CodeRabbit -->